nixos/modules/services/mail/listmonk/default.nix

```diff
@@ -0,0 +1,181 @@
+      database.createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Create the PostgreSQL database and database user locally.";
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.listmonk;
+        defaultText = lib.literalExpression "pkgs.listmonk";
+        description = lib.mdDoc "Package to use for this service";
+      };
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = tomlFormat.type;
+        };
+        description = lib.mdDoc ''Static settings set in the config.toml, see <https://github.com/knadh/listmonk/blob/master/config.toml.sample> for details.
+        You can set secrets using the secretFile option with environment variables following <https://listmonk.app/docs/configuration/#environment-variables>.'';
+      };
+      mutableDatabaseSettings = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''Database settings will be reset to the value set in this module if this is not enabled.
+        Enable this if you want to persist changes you have done in the application.
+        '';
+      };
+      databaseSettings = mkOption {
+        default = null;
+        type = with types; nullOr (submodule {
+          freeformType = oneOf [ (listOf str) (listOf (attrsOf anything)) str int bool ];
+
+          options = {
+            "app.notify_emails" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "Administrator emails for system notifications";
+            };
+
+            "privacy.exportable" = mkOption {
+              type = listOf str;
+              default = ["profile" "subscriptions" "campaign_views" "link_clicks"];
+              description = lib.mdDoc "List of fields which can be exported through an automatic export request";
+            };
+
+            "privacy.domain_blocklist" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule (import ./smtp.nix));
+              description = lib.mdDoc "List of outgoing SMTP servers";
+            };
+
+            # TODO: refine this type based on the smtp one.
+            "bounce.mailboxes" = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; oneOf [ str int bool ];
+              });
+              default = [];
+              description = lib.mdDoc "List of bounce mailboxes";
+            };
+
+            messengers = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "List of messengers, see: <https://github.com/knadh/listmonk/blob/master/models/settings.go#L64-L74> for options.";
```

Comment:
Do we have a better link than source code? Maybe a doc page?

Reply:
The doc page do not provide guidance on how to fill these fields: https://listmonk.app/docs/messengers/